app/main.ipynb

```diff
@@ -400,26 +1541,65 @@
     "result = 0\n",
     "\n",
     "%time unknown_signature(vector1, vector2)\n",
+    "unknown_signature(vector1, vector2)\n",
```

Comment:
The function 'unknown_signature' is executed twice, which is redundant. Consider removing one of the calls to avoid unnecessary computation.